### ISSUE TITLE
`globalMsaAlignment()` output differs b/n clang and gcc

### ISSUE THREAD

**Issue by temehi** [seqan/seqan#1585]
The results of globalMsaAlignment( ) are not identical among compilers. 
you can check using `demo/dox/stream/lexical_cast.cpp` if compiled with gcc the output is as follows 
```
      0     .    :    .    :    .    :    .    :    .    : 
        DPKKPRGKMSSYAFFVQTSREEHKKKHPDASVNFSEFSKKCSERWKTMSA
          | |   |          |       |    |  | ||     ||    
        RVKRP---MNAFIVWSRDQRRKMALENPRMR-NS-EISKQLGYQWKMLTE
          | |              | | |   | |  |     | |    | | |
        FPKKP---LTPYFRFFMEKRAKYAKLHPEMS-NL-DLTKILSKKYKELPE
          |||   |        | ||                  ||      |  
        HIKKP---LNAFMLYMKEMRANVVAESTLKE-SA-AINQILGRRWHALSR

     50     .    :    .    :    .    :    .    :    .    : 
        KEKGKFEDMAKADKARYEREMKTY---------------IP--PKG---E
         ||  |   |    |   || |                  |         
        AEKWPFFQEAQKLQAMH-RE-K-----YP------NYKYRPRRKAKMLPK
          |    |  |       |         |               ||   |
        KKKMKYIQDFQREKQEFERNLARFREDHP------DL--IQ--NAK---K
            ||      | |             |                    |
        EEQAKYYELARKERQLH-MQ-L-----YPGWSARDNYGKKKKRKRE---K
```
Whereas with clang:
```
      0     .    :    .    :    .    :    .    :    .    : 
        DPKKPRGKMSSYAFFVQTSREEHKKKHPDASVNFSEFSKKCSERWKTMSA
          | |   |          |       |    |  | ||     ||    
        RVKRP---MNAFIVWSRDQRRKMALENPRMR-NS-EISKQLGYQWKMLTE
          | |              | | |   | |  |     | |    | | |
        FPKKP---LTPYFRFFMEKRAKYAKLHPEMS-NL-DLTKILSKKYKELPE
          |||   |        | ||                  ||      |  
        HIKKP---LNAFMLYMKEMRANVVAESTLKE-SA-AINQILGRRWHALSR

     50     .    :    .    :    .    :    .    :    .    : 
        KEKGKFEDM-AKADKARYEREMKTY---------------IP--PKG---
         ||  |     |   |   || |                  |        
        AEKWPFFQEAQK-LQAMH-RE-K-----YP------NYKYRPRRKAKMLP
          |    |  |        |         |               ||   
        KKKMKYIQDFQR-EKQEFERNLARFREDHP------DL--IQ--NAK---
            ||       | |             |                    
        EEQAKYYELARK-ERQLH-MQ-L-----YPGWSARDNYGKKKKRKRE---

    100   
        E
         
        K
        |
        K
        |
        K
```

Does any one have a clue why this is happening? May be it's related to order of function evaluation ...?? 
I found out about this because the tests of PR #1579 were falling. This has to be solved first for  PR #1579 to pass.

**Comment by temehi** [comment]
@cpockrandt this is solved together with #1579

**Comment by cpockrandt** [comment]
@temehi Thanks for the hint, didn't notice that :see_no_evil:
